src/snakemake/dag.py

```
@@ -416,59 +417,100 @@ def update_conda_envs(self):
+                                f"{f.storage_object.is_ondemand_eligible} with access "
+                                f"patterns {','.join(map(str, file_access_patterns))}"
+                            )
+
+                            async def retrieve_and_yield():
+                                await f.retrieve_from_storage()
+                                for job in yield_after_file[f]:
+                                    await asyncio.to_thread(ready_queue.put, job)
+
+                            tg.create_task(retrieve_and_yield())
+                except ExceptionGroup as e:
+                    raise WorkflowError("Failed to retrieve input from storage.", e)
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)